log4j-core/src/main/java/org/apache/logging/log4j/core/filter/MutableThreadContextMapFilter.java

```diff
@@ -364,23 +367,29 @@ private class FileMonitor implements Runnable {
         @Override
         public void run() {
             final ConfigResult result = getConfig(source, authorizationProvider);
-            if (result.status == Status.SUCCESS) {
-                filter = ThreadContextMapFilter.createFilter(result.pairs, "or", getOnMatch(), getOnMismatch());
-                LOGGER.info("Filter configuration was updated: {}", filter.toString());
-                for (FilterConfigUpdateListener listener : listeners) {
-                    listener.onEvent();
-                }
-            } else if (result.status == Status.NOT_FOUND) {
-                if (!(filter instanceof NoOpFilter)) {
-                    LOGGER.info("Filter configuration was removed");
+            switch (result.status) {
+                case SUCCESS:
+                    filter = ThreadContextMapFilter.createFilter(result.pairs, "or", getOnMatch(), getOnMismatch());
+                    LOGGER.info("MutableThreadContextMapFilter configuration was updated: {}", filter.toString());
+                    break;
+                case NOT_FOUND:
+                    if (!(filter instanceof NoOpFilter)) {
+                        LOGGER.info("MutableThreadContextMapFilter configuration was removed");
+                        filter = new NoOpFilter();
+                    }
+                    break;
+                case EMPTY:
+                    LOGGER.debug("MutableThreadContextMapFilter configuration is empty");
                     filter = new NoOpFilter();
+                    break;
+            }
+            switch (result.status) {
+                case SUCCESS:
+                case NOT_FOUND:
+                case EMPTY:
                     for (FilterConfigUpdateListener listener : listeners) {
                         listener.onEvent();
                     }
-                }
-            } else if (result.status == Status.EMPTY) {
-                LOGGER.debug("Filter configuration is empty");
-                filter = new NoOpFilter();
             }
```

Comment:
Here we have a small modification of behavior: an `EMPTY` result before didn't call the listeners.

However I don't consider this a minor change, since the listeners are basically only useful in tests.

Reply:
If so,

1. Why don't we call listeners for all `result.status` values?
2. Can we deprecate the listeners feature?

Reply:
1. In practice the listener is used to stop the unit test at certain moment. :cry: If we were to run the listener when the config file was not modified, it would complexify the listener: the listener does not have a `Result` parameter.
2. I would say yes, we can do it in `2.25.0`.

Reply:
I added a comment explaining why we call the listeners for these statuses and not the other two.